server/odc-service/src/main/java/com/oceanbase/odc/service/connection/FileSystemConnectionTesting.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.connection;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.stereotype.Component;
+
+import com.aliyun.oss.OSSErrorCode;
+import com.aliyun.oss.OSSException;
+import com.oceanbase.odc.core.shared.PreConditions;
+import com.oceanbase.odc.core.shared.constant.ConnectType;
+import com.oceanbase.odc.plugin.connect.api.TestResult;
+import com.oceanbase.odc.service.cloud.model.CloudProvider;
+import com.oceanbase.odc.service.connection.model.ConnectionConfig;
+import com.oceanbase.odc.service.connection.model.ConnectionTestResult;
+import com.oceanbase.odc.service.objectstorage.cloud.CloudResourceConfigurations;
+import com.oceanbase.odc.service.objectstorage.cloud.client.CloudClient;
+import com.oceanbase.odc.service.objectstorage.cloud.client.CloudException;
+import com.oceanbase.odc.service.objectstorage.cloud.model.DeleteObjectsRequest;
+import com.oceanbase.odc.service.objectstorage.cloud.model.ObjectMetadata;
+import com.oceanbase.odc.service.objectstorage.cloud.model.ObjectStorageConfiguration;
+import com.oceanbase.tools.migrator.common.exception.UnExpectedException;
+
+import lombok.NonNull;
+
+/**
+ * @Author：tinker
+ * @Date: 2024/11/19 11:24
+ * @Descripition:
+ */
+
+@Component
+public class FileSystemConnectionTesting {
+
+    private static final String COS_ENDPOINT_REGEX = "cos\\.(\\w+-\\w+)\\.myqcloud\\.com";
```

Comment:
as this regex will be compiled to Pattern. declare pattern, please confirm Pattern is thread safe